### PR TITLE
Adding testability statements to the CCSM specification

### PR DESCRIPTION
**Pull request by Therecanbeonlyone1969**
…curity)

### CODE REVIEW

File: ccsm/baseline-ccsm-v1.0-psd01.md [hunk 1]
```diff
@@ -216,25 +216,37 @@ A CCSM utilized in a BPI MUST support cryptographic algorithms based on commonly
 
 The usage of cryptographic libraries that successfully passed the National Institute of Standards and Technology (NIST) Cryptographic Module Verification Program (CMVP) is recommended.
 
+[[R1]](#r1) testability: Any security-audited cryptographic library will have test-vectors, and is, therefore, testable.
+
 #### **[R2]**	
 If the CCSM utilized in a BPI utilizes a Peer-to-Peer (P2P) message protocol, the protocol MUST support end-to-end encryption.
 
+[[R2]](#r2) testability: libp2p as a very commonly used P2P message protocol supports end-to-end encryption, see [here](https://github.com/libp2p/specs/blob/master/noise/README.md) for the NOISE specification and relevant tests [here](https://github.com/libp2p/go-libp2p/blob/master/test-plans/cmd/ping/main.go). This shows that the requirement is testable.
```
Comment (Kasshern) [inline ccsm/baseline-ccsm-v1.0-psd01.md:224]: This show of testability seems to hinge upon libp2p being used. Does this not create further requirement for implementers to use libp2p? Why can we not just hint that outside protocols implementing test suites constitutes testability?

File: ccsm/baseline-ccsm-v1.0-psd01.md [hunk 2]
```diff
@@ -216,25 +216,37 @@ A CCSM utilized in a BPI MUST support cryptographic algorithms based on commonly
...
 
+[[R4]](#r4) testability: Programmatic economic security assurances are typically implemented within the CCSM's consensus protocol. Examples are Proof-of-Work for Bitcoin, and Proof-of-Stake for Ethereum, as common CCSMs. Ethereum's Proof-of-Stake protocol is specified [here](https://arxiv.org/abs/1710.09437) and tests of a security audited reference implementation can be found [here](https://github.com/prysmaticlabs/prysm/tree/develop/testing). Therefore, the requirement is testable.
```
Comment (Kasshern) [inline ccsm/baseline-ccsm-v1.0-psd01.md:236]: This feels similar to the one above.

File: ccsm/baseline-ccsm-v1.0-psd01.md [hunk 3]
```diff
@@ -216,25 +216,37 @@ A CCSM utilized in a BPI MUST support cryptographic algorithms based on commonly
...
+
 #### **[D1]**	
 The CCSM utilized in a BPI SHOULD be compatible with CCSM protocol execution in Trusted Execution Environments (TEE).
 
 *Note, a TEE is a secure area of a main processor. It guarantees code and data loaded inside to be protected with respect to confidentiality and integrity. A TEE as an isolated execution environment provides security features such as isolated execution, integrity of applications executing within the TEE, along with confidentiality of their assets.*
 
+[[D1]](#d1) testability: An example of a CCSM utilizing a TEE is Hyperledger Sawtooth with its [test suite](https://github.com/hyperledger/sawtooth-core/tree/main/cli/tests). Therefore, the requirement is testable.
```
Comment (Kasshern) [inline ccsm/baseline-ccsm-v1.0-psd01.md:243]: So is the logic that is a single example of something accomplishing the requirement exists and is testable, then the requirement itself should be able to be built out in such a way that is also testable?

**Review by Kasshern — APPROVED** [review]

File: ccsm/baseline-ccsm-v1.0-psd01.md (hunk 1 above)
Comment (Therecanbeonlyone1969) [reply ccsm/baseline-ccsm-v1.0-psd01.md:224]: an example with a test suite that meets and tests the requirement is sufficient. there does not have to be a formal standard defined with a test suite for reference implementations. One could argue abstractly but concretely is better IMHO. @Kasshern

File: ccsm/baseline-ccsm-v1.0-psd01.md (hunk 2 above)
Comment (Therecanbeonlyone1969) [reply ccsm/baseline-ccsm-v1.0-psd01.md:236]: see my comment above.

File: ccsm/baseline-ccsm-v1.0-psd01.md (hunk 3 above)
Comment (Therecanbeonlyone1969) [reply ccsm/baseline-ccsm-v1.0-psd01.md:243]: see my 1st comment

File: ccsm/baseline-ccsm-v1.0-psd01.md (hunk 1 above)
Comment (Ybittan) [reply ccsm/baseline-ccsm-v1.0-psd01.md:224]: In that case, for me as well, LGTM

**Review by Ybittan — APPROVED** [review]

**Pull request merged**